sections/futures/MarketInfoBox/MarketInfoBox.tsx

```diff
@@ -19,20 +25,82 @@ const MarketInfoBox: React.FC<MarketInfoBoxProps> = ({
 	buyingPower,
 	marginUsage,
 	isMarketClosed,
+	potentialTrade,
+	maxLeverageValue,
+	currencyKey,
 }) => {
+	const potentialTradeDetails = useGetFuturesPotentialTradeDetails(
+		currencyKey as CurrencyKey,
+		potentialTrade
+	);
+
+	const previewTrade = potentialTradeDetails.data ?? null;
+
+	const previewTradeData = React.useMemo(() => {
+		const size = wei(potentialTrade?.size || zeroBN);
+		const potentialAvailableMargin =
+			previewTrade?.margin?.sub(previewTrade?.minInitialMargin) ?? zeroBN;
```

Comment:
`minInitialMargin` doesn't change based on the position side (long/short) so this isn't correct. It looks like the value is always positive, but some trades will actually increase available margin (long while having an open short position)

From our discussion: `total margin - (notional value * max leverage)`

So the formula would look more like (with better validation):
`previewTrade.margin.sub(potentialTrade.size.mul(maxLeverageValue))`

Reply:
See above comment on "max leverage" and how this is incorrect.